src/statd/python/yanger/yanger.py

```diff
@@ -63,11 +63,23 @@ def json_get_yang_origin(addr):
 
 
 def getitem(data, key):
-    """Get sub-object from an object"""
-    while key:
-        data = data[key[0]]
-        key = key[1:]
+    """Get sub-object from an object safely."""
+    if data is None:
+        return None
 
+    for k in key:
+        try:
+            if isinstance(data, dict):
```

Comment:
squeese these two to a oneliner:
if isinstance(data, dict) or (isinstance(data, list) and isinstance(k, int))